statics/js/components/rule-detail.js

```diff
@@ -0,0 +1,494 @@
+  */
+  BridgeLayout.prototype.portname = function(pr) {
+    var p = (typeof (pr) === 'string') ? parseInt(pr) : pr;
+    if (p === ANY_PORT) return 'ANY';
+    if (p === LOCAL_PORT) return 'LOCAL';
+    if (p === SAME_PORT || p === undefined) return '';
+    var nodes = this.interfaces[p];
+    if (nodes === undefined) return '???';
+    var port = nodes[1];
+    var portname = (port === undefined) ? '???' : port.metadata.Name;
+    return portname;
+  };
+
+  /** Check if rule is highlighted. */
+  BridgeLayout.prototype.isHighlighted = function(rule) {
+    var current = this.store.state.currentRule;
+    var status = current && rule.UUID === current.metadata.UUID;
+    return status;
+  };
+
+  return BridgeLayout;
+}());
+
+/** Graphical component that represents a set of rules associated to a given
+ *  Openflow table and port and displayed as a single HTML table.
+ */
+Vue.component('rule-table-detail', {
+  template: '\
+    <div class="dynamic-table">\
+      <table class="table table-bordered table-condensed">\
+        <thead>\
+            <tr>\
+                <th class="filters-column">filters</th>\
+                <th class="summary-column">summary</th>\
+                <th class="actions-column">actions</th>\
+            </tr>\
+        </thead>\
+        <tbody>\
+            <tr v-for="rule in rules"\
+                :id="\'R-\' + rule.UUID"\
+                v-bind:class="{soft: layout.isHighlighted(rule)}">\
+                <td>{{rule.filters}}</td>\
+                <td>\
+                    <table>\
+                        <tr v-for="act in rule.outAction">\
+                            <td><img :src="layout.imgsrc(act.action)"\
+                                    width="16"\
+                                    height="16" /></td>\
+                            <td v-on:mouseover="layout.mark(act.port)"\
+                                v-on:mouseleave="layout.unmark(act.port)"\
+                                v-on:click="layout.select(act.port)">\
+                                <span class="port-link">{{layout.portname(act.port)}}</span>\
+                            </td>\
+                            <td> <a class="table-link" v-on:click="layout.switchTab(act.table)">{{act.table}}</a></td>\
+                        </tr>\
+                    </table>\
+                </td>\
+                <td>{{rule.actions}}</td>\
+            </tr>\
+        </tbody>\
+      </table>\
+    </div>',
+  props: {
+    rules: {
+      type: Object,
+      required: true
+    },
+    layout: {
+      type: Object,
+      required: true
+    }
+  },
+});
+
+/** Vue component showing the rules associated to a bridge */
+Vue.component('rule-detail', {
+  template: '\
+<div class="rules-detail flow-ops-panel" v-if="Object.keys(layout.structured).length > 0">\
+    <ul class="nav nav-pills"\
+        role="tablist">\
+        <li class="nav-header"> <a>Table </a></li>\
```

Comment:
Then it must be done for ports too. I think it is a little heavy... Let us try again. The new code is more in the spirit of main menus with an explicit header. I used container-fluid so that we can have multiple lines when there are a lot of tables or ports.